File: GLLara/GLLLightController.m
//
//  GLLLightController.m
//  GLLara
//
//  Created by Torsten Kammer on 13.01.13.
//  Copyright (c) 2013 Torsten Kammer. All rights reserved.
//

#import "GLLLightController.h"

@implementation GLLLightController

- (id)initWithLight:(NSManagedObject *)light;
{
    if (!(self = [super init])) return nil;
    
	_light = light;
	
	return self;
}

- (id)representedObject
{
	return self.light;
}

#pragma mark - Outline view data source

- (id)outlineView:(NSOutlineView *)outlineView child:(NSInteger)index ofItem:(id)item
{
	return nil;
}

- (id)outlineView:(NSOutlineView *)outlineView objectValueForTableColumn:(NSTableColumn *)tableColumn byItem:(id)item
{
	if ([self.light.entity.name isEqual:@"GLLAmbientLight"])
		return NSLocalizedString(@"Ambient", @"source view - lights");
	else
		return [NSString stringWithFormat:NSLocalizedString(@"Diffuse %@", @"source view - lights"), [self.light valueForKey:@"index"]];
}

- (BOOL)outlineView:(NSOutlineView *)outlineView isItemExpandable:(id)item
{
	return NO;
}

- (NSInteger)outlineView:(NSOutlineView *)outlineView numberOfChildrenOfItem:(id)item
{
	return 0;
}

@end

File: GLLara/GLLLightController.swift
//
//  GLLLightController.m
//  GLLara
//
//  Created by Torsten Kammer on 13.01.13.
//  Copyright (c) 2013 Torsten Kammer. All rights reserved.
//

import Cocoa

/**
 * Source list controller for a single light (whether diffuse or ambient).
 */
@objc class GLLLightController : NSObject, NSOutlineViewDataSource {
    init(light: NSManagedObject, parentController: AnyObject) {
        self.light = light
        self.parentController = parentController
    }
    
    let light: NSManagedObject
    @objc var representedObject: NSManagedObject {
        return light
    }
    @objc weak var parentController: AnyObject?
    
    func outlineView(_ outlineView: NSOutlineView, child index: Int, ofItem item: Any?) -> Any {
        return "" // Should never come here
    }
    
    func outlineView(_ outlineView: NSOutlineView, objectValueFor tableColumn: NSTableColumn?, byItem item: Any?) -> Any? {
        if light.entity.name == "GLLAmbientLight" {
            return NSLocalizedString("Ambient", comment: "source view - lights");
        } else {
            return String(format: NSLocalizedString("Diffuse %@", comment: "source view - lights"), light.value(forKey: "index") as! NSNumber)
        }
    }
    
    func outlineView(_ outlineView: NSOutlineView, numberOfChildrenOfItem item: Any?) -> Int {
        return 0
    }
    
    func outlineView(_ outlineView: NSOutlineView, isItemExpandable item: Any) -> Bool {
        return false
    }
}
